mqt/qmap/subarchitectures.py

```diff
@@ -0,0 +1,275 @@
+"""
+Functionality for computing good subarchitectures for quantum circuit mapping.
+
+The function provided in this module... TODO
+"""
+from __future__ import annotations
+
+import os
+import pickle
+from collections import defaultdict
+from itertools import combinations
+from typing import Dict, List, NewType, Tuple, Union
+
+import networkx as nx
```

Comment:
Is networkx really needed here?

Reply:
I guess mostly for convenience for converting edgelists to retworkx graphs. But I can do away without.

Reply:
I see. Doesn't `retworkx` provide such a functionality?
It seems a little bit wasteful to add a whole dependency for such little functionality. Although, overall, I would be fine with including it anyway.

Reply:
Since it takes just 3 LOC more, it isn't a big deal.